packages/dds/sequence/src/intervals/sequenceInterval.ts

```diff
@@ -101,10 +101,23 @@ function maxSide(sideA: Side, sideB: Side): Side {
  * `mergeTreeReferencesCanSlideToEndpoint` feature flag set to true, the endpoints
  * of the interval that are exclusive will have the ability to slide to these
  * special endpoint segments.
- * @legacy
  * @alpha
+ * @legacy
  */
-export class SequenceInterval implements ISerializableInterval {
+export interface SequenceInterval extends ISerializableInterval {
+	readonly start: LocalReferencePosition;
+	/**
+	 * End endpoint of this interval.
+	 * @remarks This endpoint can be resolved into a character position using the SharedString it's a part of.
+	 */
+	readonly end: LocalReferencePosition;
+	readonly intervalType: IntervalType;
+	readonly startSide: Side;
+	readonly endSide: Side;
+	readonly stickiness: IntervalStickiness;
+}
+
+export class SequenceIntervalClass implements SequenceInterval {
```

Comment:
This should have an explicit tsdoc and `@internal` tag. No complaints from the tooling?

Reply:
no it shouldn't, it's never exported from the package